app.js

```diff
@@ -1,72 +1,98 @@
 
-app.get('/download/:id', function(req, res) {
-  res.sendFile(path.join(__dirname + '/public/download.html'));
+app.get("/download/:id", function(req, res) {
+  res.sendFile(path.join(__dirname + "/public/download.html"));
 });
 
-app.get('/assets/download/:id', function(req, res) {
+app.get("/assets/download/:id", function(req, res) {
   
   let id = req.params.id;
   client.hget(id, "filename", function(err, reply) { // maybe some expiration logic too
-  if (!reply) {
+    if (!reply) {
       res.sendStatus(404);
     } else {
-      res.setHeader('Content-Disposition', 'attachment; filename=' + reply);
-      res.setHeader('Content-Type', 'application/octet-stream');
+      res.setHeader("Content-Disposition", "attachment; filename=" + reply);
+      res.setHeader("Content-Type", "application/octet-stream");
       
-      res.download(__dirname + '/static/' + id, reply, function(err) {
+      res.download(__dirname + "/static/" + id, reply, function(err) {
         if (!err) {
           client.del(id);
-          fs.unlinkSync(__dirname + '/static/' + id);
+          fs.unlinkSync(__dirname + "/static/" + id);
         }
       });
     }
   })
   
 });
 
-app.route('/upload/:id')
-    .post(function (req, res, next) {
+app.post("/delete/:id", function(req, res) {
+  let id = req.params.id;
+  let delete_token = req.body.delete_token;
+  
+  if (!delete_token){ 
+    res.sendStatus(404);
+  }
+
+  client.hget(id, "delete", function(err, reply) {
+    if (!reply) {
+      res.sendStatus(404);
+    } else {
+      client.del(id);
+      fs.unlinkSync(__dirname + "/static/" + id);
+      res.sendStatus(200);
+    }
+  })
+});
+
+app.post("/upload/:id", function (req, res, next) {
 
         var fstream;
         req.pipe(req.busboy);
-        req.busboy.on('file', function (fieldname, file, filename) {
+        req.busboy.on("file", function (fieldname, file, filename) {
             console.log("Uploading: " + filename);
 
             //Path where image will be uploaded
-            fstream = fs.createWriteStream(__dirname + '/static/' + req.params.id);
+            fstream = fs.createWriteStream(__dirname + "/static/" + req.params.id);
             file.pipe(fstream);
-            fstream.on('close', function () {  
+            fstream.on("close", function () {  
                 let id = req.params.id;
-                client.hset(id, "filename", filename, redis.print);
-                client.hset(id, "expiration", 0, redis.print);
+                let uuid = Math.floor(Math.random() * 10000000).toString()
+                         + Math.floor(Math.random() * 10000000).toString()
+                         + Math.floor(Math.random() * 10000000).toString();
+
+                client.hset(id, "filename", filename);
+                client.hset(id, "expiration", 0);
+                client.hset(id, "delete", uuid);
+
+                // delete the file off the server in 24 hours
+                setTimeout(function() {
```

Comment:
It's not ideal to have a bunch of timeout callbacks hanging around like this. We'll need to find another way to do this before launch. Maybe we can rely on expiring them in S3, otherwise a separate process from the server to do the cleanup would be ok.